src/wavesurfer.ts

```diff
@@ -186,12 +186,15 @@ class WaveSurfer extends Player<WaveSurferEvents> {
   }
 
   private initPlayerEvents() {
+    const timeUpdateRerender = () => {
+      const currentTime = this.getCurrentTime()
+      this.renderer.renderProgress(currentTime / this.getDuration(), this.isPlaying())
+      this.emit('timeupdate', currentTime)
+      requestAnimationFrame(timeUpdateRerender)
```

Comment:
This would create an infinite loop here, wouldn't it?

You probably want to wrap only this part:
```js
requestAnimationFrame(() => {
  this.renderer.renderProgress(this.getCurrentTime() / this.getDuration(), this.isPlaying())
})
```

Reply:
Based on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) we need to fire it every single time because it's one shot. I've added the `isPlaying()` check, so this is not fired everytime.

Reply:
It's one-shot but it calls itself.